hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsRestOperation.java

```diff
@@ -207,12 +210,24 @@ public void execute() throws AzureBlobFileSystemException {
    * fails, there may be a retry.  The retryCount is incremented with each
    * attempt.
    */
-  private boolean executeHttpOperation(final int retryCount) throws AzureBlobFileSystemException {
+  private boolean executeHttpOperation(final int retryCount, TracingContext tracingContext)
+          throws AzureBlobFileSystemException {
     AbfsHttpOperation httpOperation = null;
     try {
       // initialize the HTTP request and open the connection
       httpOperation = new AbfsHttpOperation(url, method, requestHeaders);
       incrementCounter(AbfsStatistic.CONNECTIONS_MADE, 1);
+      if (client.isCorrelationHeaderEnabled()) {
+        tracingContext.generateClientRequestID();
+        httpOperation.getConnection()
+                .setRequestProperty(HttpHeaderConfigurations.X_MS_CLIENT_REQUEST_ID,
+                        tracingContext.toString());
```

Comment:
Logic to return only clientReqId or all fields can be decided by TracingContext, if the config value was provided to it at init. Move line 221 and 222 to private method updateClientRequestIdHeader().

Reply:
Done